lib/Doctrine/ORM/Query/AST/TypedExpression.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Query\AST;
+
+use Doctrine\DBAL\Types\Type;
+
+interface TypedExpression
```

Comment:
Is this interface for internal use only? If yes, please mark it as such with `@internal` + a comment. If not, please document it.

Reply:
I don't think so it can be used by custom function created by the user (for instance COS() could implement this interface)

Reply:
OK, then maybe document it?